src/ota_metadata/legacy2/metadata.py

```diff
@@ -76,6 +78,20 @@
 DB_TIMEOUT = 16  # seconds
 
 
+def check_base_filetable(db_f: StrOrPath | None) -> StrOrPath | None:
+    if not db_f or not Path(db_f).is_file():
+        return
+
+    with contextlib.suppress(Exception), contextlib.closing(
+        sqlite3.connect(f"file:{db_f}?mode=ro&immutable=1", uri=True)
+    ) as con:
+        if utils.check_db_integrity(
+            con,
+            table_name=FileTableRegularORM.orm_bootstrap_table_name,
+        ):
+            return db_f
```

Comment:
We ensure that the delta_source's file_table database is valid by checking the corresponding table(`ft_regular`) is presented and valid.